File: benchmarks/live_suite/matmul.py
def run_matmul(n):
    a = [0] * (n * n)
    b = [0] * (n * n)
    c = [0] * (n * n)

    for i in range(n):
        for j in range(n):
            a[i * n + j] = (i + j) % 100
            b[i * n + j] = (i * j) % 100

    for i in range(n):
        for k in range(n):
            a_ik = a[i * n + k]
            for j in range(n):
                c[i * n + j] += a_ik * b[k * n + j]

    return c[n * n - 1]

if __name__ == "__main__":
    checksum = 0
    for _ in range(2): # 2 reps for Python
        checksum = run_matmul(128)
    print(f"Matmul Checksum: {checksum}")

File: bench/live_suite/matmul.go
package main

import "fmt"

func runMatmul(n int) int64 {
	a := make([]int64, n*n)
	b := make([]int64, n*n)
	c := make([]int64, n*n)

	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			a[i*n+j] = int64((i + j) % 100)
			b[i*n+j] = int64((i * j) % 100)
		}
	}

	for i := 0; i < n; i++ {
		for k := 0; k < n; k++ {
			aIK := a[i*n+k]
			for j := 0; j < n; j++ {
				c[i*n+j] += aIK * b[k*n+j]
			}
		}
	}

	return c[n*n-1]
}

func main() {
	var checksum int64
	for iter := 0; iter < 10; iter++ {
		checksum = runMatmul(128)
	}
	fmt.Printf("Matmul Checksum: %d\n", checksum)
}
